trinity/protocol/common/proto.py

```diff
@@ -0,0 +1,53 @@
+
+
+class ChainProtocol(Protocol, ABC):
+
+    @classmethod
+    def configure_protocol(
+            cls,
+            headerdb: BaseAsyncHeaderDB,
+            network_id: int,
+            vm_configuration: Tuple[Tuple[int, Type[BaseVM]], ...]) -> Type[ChainProtocol]:
+        return type(
+            f'{cls.__name__}Network{network_id}',
+            (cls, ),
+            dict(
+                headerdb=headerdb,
+                network_id=network_id,
+                vm_configuration=vm_configuration,
+            ),
+        )
+
+    @abstractmethod
+    @property
+    def headerdb(self):
+        pass
+
+    @abstractmethod
+    @property
+    def network_id(self):
+        pass
+
+    @abstractmethod
+    @property
+    def vm_configuration(self):
+        pass
+
+    @property
+    async def genesis(self) -> BlockHeader:
```

Comment:
Just noting that I'd like to change this to a method as I find awaitable properties to be confusing and thus prone to errors.